crates/autopilot/src/domain/fee/mod.rs

```diff
@@ -55,13 +61,34 @@ impl ProtocolFee {
             }
         }
 
-        let protocol_fees = match &self.policy {
-            policy::Policy::Surplus(variant) => variant.apply(&order, quote),
-            policy::Policy::PriceImprovement(variant) => variant.apply(&order, quote),
-            policy::Policy::Volume(variant) => variant.apply(&order),
-        }
-        .into_iter()
-        .collect_vec();
+        let order_ = boundary::Amounts {
+            sell: order.data.sell_amount,
+            buy: order.data.buy_amount,
+            fee: order.data.fee_amount,
+        };
+        let quote_ = boundary::Amounts {
+            sell: quote.sell_amount,
+            buy: quote.buy_amount,
+            fee: quote.fee,
+        };
+        let protocol_fees = protocol_fees
+            .iter()
+            // TODO: support multiple fee policies
+            .find_map(|fee_policy| {
+                let outside_market_price = boundary::is_order_outside_market_price(&order_, &quote_);
```

Comment:
It doesn't change much logically and is mainly style, but I think this code would be simpler if instead of a single config value which is a vector of an enum, we use two optional fields `in_market`, `out_of_market` (can be combined on a struct for passing around into the domain)

Reply:
ditto: https://github.com/cowprotocol/services/pull/2550#discussion_r1536770540

Reply:
The idea of having a single CLI argument that specifies a list of policies that apply to orders of a certain variant is not tied to the representation in the domain, though.
We could still have the single CLI argument (which seems nicer IMO but I might be the only one) and still pass around fee policies not wrapped in an enum and instead with an additional marker field.